bonxai_core/include/bonxai/bonxai.h

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Contributors to the Bonxai Project
+ * Copyright Contributors to the OpenVDB Project
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#ifdef _MSC_VER
+#define BONXAI_EXPORT __declspec(dllexport)
+#else
+#define BONXAI_EXPORT __attribute__((visibility("default")))
+#endif
+
+#ifdef __cplusplus
+#include <cstddef>
+#include <cstdint>
+#define BONXAI_CALL extern "C"
+#else
+#include <stddef.h>
+#include <stdint.h>
+#define BONXAI_CALL
+#endif
+
+#define BONXAI_API BONXAI_EXPORT BONXAI_CALL
+
+#define BONXAI_OK 0
+#define BONXAI_ERR_INV_ARG 1
+
+#define BONXAI_FALSE 0
+#define BONXAI_TRUE 1
```

Comment:
why?

Reply:
As bool's and enum's are not exactly ideal for language interop scenarios because of unportable ABI an int type is usually employed, and if you have your boolean alias you might as well have fitting constants. These can be recognized by interop tools.